src/modules/27-platform/auth-settings/pages/Configuration/EnablePublicAccess/__tests__/EnablePublicAccess.test.tsx

```diff
@@ -27,17 +27,21 @@ describe('EnablePublicAccess', () => {
     })
     ;(setPublicAccessPromise as jest.Mock).mockImplementationOnce(mockSetPublicAccessPromise)
 
-    const { getByTestId, queryByText } = render(
+    const { getByTestId } = render(
       <TestWrapper>
         <EnablePublicAccess enabled={false} refetchAuthSettings={refetchAuthSettings} canEdit={true} />
       </TestWrapper>
     )
 
     const switchBtn = getByTestId('toggle-enable-public-access')
     await userEvent.click(switchBtn!)
-    await waitFor(() => expect(setPublicAccessPromise).toBeCalled())
 
-    expect(queryByText('platform.authSettings.updatedPublicAccess')).toBeTruthy()
+    const dialogContainer = findDialogContainer()
+    expect(dialogContainer).toBeTruthy()
+    const enableButton = queryByTextFromRTL(dialogContainer as HTMLElement, 'enable')
```

Comment:
Try to use `getByRole` instead of by text if possible. It's closer to how a user perceives the application.
```suggestion
    const enableButton = getByRole(dialogContainer as HTMLElement, 'button', { name: 'enable' })
```

Reply:
Thanks for the suggestion. Done.